elements/pfe-icon/src/pfe-icon.scss

```diff
@@ -27,143 +13,89 @@ $pfe-icon--border-style:                     map-get($pfe-vars, border-style);
+  --pfe-icon--BorderColor:             transparent;
+  --pfe-icon--Padding:                 .05em;
 
   display: inline-block;
   vertical-align: middle;
-  @include pfe-icon-set-svg-size(1em);
   border-radius: 50%;
-  background: var(--pfe-icon--color--bg);
-  border: $pfe-icon--border-width $pfe-icon--border-style var(--pfe-icon--color--border);
+  background-color: #{pfe-local(BackgroundColor)};
+  border: #{pfe-var(ui--border-width)} #{pfe-var(ui--border-style)} #{pfe-local(BorderColor)};
+
+  @include pfe-icon-set-svg-size(1em);
 
   svg {
-    fill: var(--pfe-broadcasted--color--text);
+    fill: #{pfe-apply-broadcast(text)};
   }
 
   @at-root #{&}([data-block]) {
     display: block;
-    margin-bottom: var(--pfe-icon--spacing);
-    margin-top: var(--pfe-icon--spacing);
+    margin-bottom: #{pfe-local(spacing)};
+    margin-top: #{pfe-local(spacing)};
 
     &:first-child {
       margin-top: 0;
     }
   }
 }
 
-:host([size="2x"]) {
-  @include pfe-icon-set-svg-size(2em);
-}
-
-:host([size="3x"]) {
-  @include pfe-icon-set-svg-size(3em);
-}
-
-:host([size="4x"]) {
-  @include pfe-icon-set-svg-size(4em);
+@for $size from 1 through 4 {
+  :host([size="#{$size}x"]) {
+    @include pfe-icon-set-svg-size(#{$size}em);
+  }
 }
 
 // TODO: Sort out sizing
-:host([size="xl"]) {
-  @include pfe-icon-set-svg-size(100px);
-}
-
-:host([size="lg"]) {
-  @include pfe-icon-set-svg-size(64px);
-}
-
-:host([size="md"]) {
-  @include pfe-icon-set-svg-size(32px);
-}
-
-:host([size="sm"]) {
-  @include pfe-icon-set-svg-size(14px);
-}
-
-:host([color="base"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(ui-base)};
-}
+$abstract-sizes: (
+  xl: 100px,
+  lg: 64px,
+  md: 32px,
+  sm: 18px
+);
 
-:host([color="complement"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(ui-complement)};
-}
-
-:host([color="accent"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(ui-accent)};
-}
-
-:host([color="critical"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(feedback--critical)};
-}
-
-:host([color="important"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(feedback--important)};
-}
-
-:host([color="moderate"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(feedback--moderate)};
-}
-
-:host([color="success"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(feedback--success)};
+// TODO: Sort out sizing
+@each $label, $size in $abstract-sizes {
+  :host([size="#{$label}"]) {
+    @include pfe-icon-set-svg-size($size);
+  }
 }
 
-:host([color="info"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(feedback--info)};
+@each $color in (base, complement, accent) {
+  :host([color="#{$color}"]) {
+    @include pfe-set-broadcast($color: #{$color}, $broadcast: text, $color-set: ui);
+  }
 }
 
-:host([color="default"]) {
-  --pfe-broadcasted--color--text:      #{pfe-color(feedback--default)};
+@each $color in (critical, important, moderate, success, info, default) {
```

Comment:
@starryeyez024 This is an example of where the $color-set can come into play.  The pfe-icon invokes the feedback color set instead of the ui color set.